### PR TITLE
Nyomi mackey-WIP

### PR DESCRIPTION
**Pull request by nyomilm**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,14 +29,23 @@ Demonstrate your understanding of this week's concepts by answering the followin
 Edit this document to include your answers after each question. Make sure to leave a blank line above and below your answer so it is clear and easy to read by your project manager
 
 1. Describe the biggest difference between `.forEach` & `.map`.
```
Comment (AsaOfDiamonds) [inline README.md:31]: .forEach mutates the original array, map creates a new array

File: README.md [hunk 2]
```diff
@@ -29,14 +29,23 @@ Demonstrate your understanding of this week's concepts by answering the followin
...
+.map returns an array, .forEach does not return.
 
 2. What is the difference between a function and a method?
+A function returns a value but a method is a part of a class.
```
Comment (AsaOfDiamonds) [inline README.md:35]: A method is a function housed within an object

File: README.md [hunk 3]
```diff
@@ -29,14 +29,23 @@ Demonstrate your understanding of this week's concepts by answering the followin
...
+
 
 3. What is closure?
+Closure is where an inner function has access to the outer function's variables. 
 
 4. Describe the four rules of the 'this' keyword.
```
Comment (AsaOfDiamonds) [inline README.md:41]: Window - default binding of the 'this' keyword
Implicit - method invocation binding of 'this' happens automatically with objects and methods
Explicit - uses function methods: call, apply, bind to explicitly bind 'this' to a function
New - used to create new objects and pass 'this' into constructor functions

File: challenges/objects-arrays.js [hunk 4]
```diff
@@ -97,23 +145,45 @@ The zoos need a list of all their animal's names (names only, not scientific) co
...
+  return animals.population < 5;
+});
+
+console.log(lowerPopulation.length);
 
 /* Request 4: .reduce() 
 
 The zoos need to know their total animal population across the United States.  Find the total population from all the zoos using the .reduce() method.
 
 */
```
Comment (AsaOfDiamonds) [inline challenges/objects-arrays.js:171]: here is a simpler way to use reduce to solve this 
const populationTotal = zooAnimals.reduce((animal, totalPopulation) => {
  return animal + totalPopulation.population;
}, 0);
console.log(populationTotal);

remember advanced array methods remove the need of using a for loop

File: challenges/prototypes.js [hunk 5]
```diff
@@ -5,29 +5,47 @@
...
+  return `${this.length * this.width * this.height}`;
+}; 
 
 /* == Step 3: Surface Area Method ==
   Create another method using CuboidMaker's prototype that returns the surface area of a given cuboid's length, width, and height. 
 
   Formula for cuboid surface area of a cube: 2 * (length * width + length * height + width * height)
 */
-
+CuboidMaker.prototype.surfaceArea = function surfaceArea() {
```
Comment (AsaOfDiamonds) [inline challenges/prototypes.js:30]: make sure to read everything, you left off part of the equation and that is why the result was not the same 
Formula for cuboid surface area of a cube: 2 * (length * width + length * height + width * height)
so you should have this
return `Surface area: ${2 * (this.length * this.width + this.length * this.height + this.width * this.height)}`;

**Review by AsaOfDiamonds — APPROVED** [review]
fantastic job on your sprint challenge, I left some comments in your code make sure you read them

File: challenges/prototypes.js (hunk 5 above)
Comment (nyomilm) [reply challenges/prototypes.js:30]: got it, fixed it!

File: challenges/objects-arrays.js (hunk 4 above)
Comment (nyomilm) [reply challenges/objects-arrays.js:171]: got it

File: README.md (hunk 3 above)
Comment (nyomilm) [reply README.md:41]: got it

File: README.md (hunk 2 above)
Comment (nyomilm) [reply README.md:35]: got it

File: README.md (hunk 1 above)
Comment (nyomilm) [reply README.md:31]: got it

**Pull request merged**